compiler/rustc_passes/src/reachable.rs

```diff
@@ -1,9 +1,15 @@
-// Finds items that are externally reachable, to determine which items
-// need to have their metadata (and possibly their AST) serialized.
-// All items that can be referred to through an exported name are
-// reachable, and when a reachable thing is inline or generic, it
-// makes all other generics or inline functions that it references
-// reachable as well.
+//! Finds local items that are externally reachable, to determine which items
+//! need to have their metadata (and possibly their AST) serialized.
+//!
+//! This set is *not* transitively closed, i.e., in general the set only contains definitions that
+//! can be reached *directly* via an exported name, not private functions that can only be reached
+//! transitively.
```

Comment:
Why would "can the item be inlined" be relevant for reachability? That's an artifact of monomorphization and our codegen strategy, but here we are pre-monomorphization. So there's some obvious notion of "can this item be called from the outside", and there's an obvious graph of items mentioning other items. When I hear "reachable" I would intuitively expect that set to be closed under the edges of that graph. The point of this comment is to make it clear that that's not the intent of this pass. (If A is reachable, and from A I can reach B, then clearly B is reachable.)

I am not entirely following what you are saying here. Do you have a concrete suggestion?

Reply:
Hm... I'm a bit worried you might suffer from the "curse of knowledge" and so have very different expectations from people that didn't think hard about how inlining and monomorphization interact. But I'll try to adjust the docs to use your framing, and then maybe add a paragraph on how that compares to other relations one might expect this to compute.

Reply:
If we want such comments to be accessible, we should really not use the term "inlined" so loosely. We're referring to a specific behavior here, which is known in various parts of the compiler as `InstantiationMode::LocalCopy` (I'd call this LocalCopy codegen in English) or `generates_cgu_internal_copy` which is basically English already. I would much prefer we use that terminology instead of "inlined".

I tried to name the query/feature that I added `cross_crate_inlinable` because that is the observable effect that it is designed to produce. If that term is muddying things, we should just rename it.

Reply:
I've certainly seen confusing comments in the compiler where the English text spoke about "inlinable" and then the code called `generates_cgu_internal_copy` and it was not clear that they were talking about the same thing... so yeah we should stick to one term and use it throughout.